debezium-core/src/main/java/io/debezium/pipeline/txmetadata/TransactionMonitor.java

```diff
@@ -129,7 +129,9 @@ public void transactionComittedEvent(OffsetContext offset) throws InterruptedExc
         if (!connectorConfig.shouldProvideTransactionMetadata()) {
             return;
         }
-        endTransaction(offset);
+        if (offset.getTransactionContext().isTransactionInProgress()) {
+            endTransaction(offset);
+        }
         offset.getTransactionContext().endTransaction();
```

Comment:
Should this go into the if block too?

Reply:
I left the last line out because its a state reset which I think we want to do regardless if we emit a commit event or not.

Reply:
Ok, I see.